### PR TITLE
DOC: Add "Layout" section to "Developer Guide / MRML Overview / Advanced topics"

### PR DESCRIPTION
**Pull request by jcfr**
* [ ] Migrate content of https://www.slicer.org/wiki/Documentation/Nightly/Developers/Layouts
* [ ] Document relationship of Camera and 3D view after https://github.com/Slicer/Slicer/pull/5410 is integrated

### CODE REVIEW

File: Docs/developer_guide/mrml_overview.md [hunk 1]
```diff
@@ -117,3 +117,84 @@ Another view of [VTK/MRML pipeline for the 2D slice views](https://www.slicer.or
...
+
+* Developer may register additionial layout.
+
+[qSlicerLayoutManager-apidoc]: http://apidocs.slicer.org/master/classqSlicerLayoutManager.html
+[vtkMRMLLayoutNode-apidoc]: http://apidocs.slicer.org/master/classvtkMRMLLayoutNode.html
+[vtkMRMLLayoutLogic-apidoc]: http://apidocs.slicer.org/master/classvtkMRMLLayoutLogic.html
+
+#### Registering a custom layout
+
+_TBD_
```
Comment (lassoan) [inline Docs/developer_guide/mrml_overview.md:142]: You can find some material in the script repository. Maybe you can add a link to that.

**Review by lassoan — APPROVED** [review]
Looks great!

File: Docs/developer_guide/mrml_overview.md [hunk 2]
```diff
@@ -117,3 +117,84 @@ Another view of [VTK/MRML pipeline for the 2D slice views](https://www.slicer.or
...
+
+* It is associated with a [vtkMRMLLayoutNode][vtkMRMLLayoutNode-apidoc] describing the current layout configuration and
+  ensuring it can be saved and restored.
+
+* It owns an instance of [vtkMRMLLayoutLogic][vtkMRMLLayoutLogic-apidoc] that controls the layout node and the view
+  nodes in a MRML scene.
+
+* Pre-defined layouts are described using XML and are registered in `vtkMRMLLayoutLogic::AddDefaultLayouts()`.
+
+* Developer may register additionial layout.
```
Comment (jcfr) [inline Docs/developer_guide/mrml_overview.md:134]:
```suggestion
Developer may register additional layout.
```

File: Docs/developer_guide/mrml_overview.md (hunk 1 above)
Comment (lassoan) [inline Docs/developer_guide/mrml_overview.md:142]:
```suggestion
See [example in the script repository](script_repository.md#customize-view-layout).
```

**Pull request merged**